docs/trace/data-prepper.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: default
+title: Data Prepper
+parent: Trace Analytics
+nav_order: 20
+---
+
+# Data Prepper
+
+Data Prepper is an independent component, not an Elasticsearch plugin, that converts data for use with Elasticsearch.
+
+
+## Install Data Prepper
+
+To use the Docker image, pull it like any other image:
+
+```bash
+docker pull <>
+```
+
+Otherwise, [download](https://opendistro.github.io/for-elasticsearch/downloads.html) the appropriate TAR file for your operating system and unzip it.
+
+
+## Configure pipelines
+
+To use Data Prepper, you define pipelines in a configuration YAML file. Each pipeline is a combination of a source, a buffer, zero or more processors, and one or more sinks:
+
+```yml
+sample-pipeline:
+  workers: 4 # the number of workers
+  delay: 100 # in milliseconds, how often the workers should run
```

Comment:
the comment might be little wrong - the delay is effectively the timeout for buffer before giving up on waiting to read.

Reply:
I should add these to the reference, as well. Do we have any recommendations around worker count for hardware type (e.g. "Don't have more workers than CPU cores on your machine") or workload (e.g. "Have at least one worker for every X documents per second")? Is each worker effectively a thread? Any limit?

Reply:
We have an item to default the workers (if not provided) to number of CPU cores (is not finished yet). To provide best recommendations we would have to finish our stress testing which would help us with accurate data but as of now I do not effectively see any blockers (obviously do not expect users to provide insanely large value for this). 

each worker effectively is a thread - we currently do not have any limits [Pending on us, need to add those limits]

Reply:
Gotcha, thanks. I think I've addressed the initial comment about the delay property, but let me know if I missed the mark. It seems like the difference in language is "workers don't run every 100ms, because it takes some them amount of time to check the buffer, and if there's no work to do, they don't check again for 100ms." So I've rephrased as "how long workers wait between read attempts."